src/main/scala/io/iohk/praos/crypto/SignerImpl.scala

```diff
@@ -0,0 +1,22 @@
+package io.iohk.praos.crypto
+
+import akka.util.ByteString
+
+
+object SignerImpl extends Signer {
+  override def signedWith[T <: Serializable](data: T, privateKey: Key): Signed[T] = {
+    // FIXME: Hasher, cipher and signature provider are hard-coded for simplicity.
+    val publicKey = getPublicKeyFromPrivateKey(privateKey)
+    val hasher = PredefinedHasher("MD5")
+    val serializedData = ByteString(data.toString)
+    val hashedData = hasher.hash(serializedData)
+    val encryptedData = CipherStubImpl.encryptWith(hashedData, publicKey)
+    val signature = SignatureProviderImpl.getSignature(encryptedData, privateKey)
```

Comment:
Could we instead `SignatureProviderImpl.getSignature(serializedData, privateKey)`? Maybe I'm missing something but if this hashing and encryption is needed maybe it should be better for it to be done on the `SignatureProvider`.

Reply:
As discussed with Nico, `SignatureProvider.splitSignature` will be renamed to `SignatureProvider.getPublicKeyFromSignature` and return only the public key. Agreed that this class will eventually disappear.